finagle-core/src/main/scala/com/twitter/finagle/Dtab.scala

```diff
@@ -154,6 +162,106 @@ object Dentry {
       d1.dst.simplified == d2.dst.simplified
     )
   }
+
+  /**
+   * A Prefix comprises a [[Path]]-matching expression.
+   *
+   * Each element in a prefix may be either a [[Label]] or [[AnyElem]].
+   * When matching a [[Path]], Label-elements must match exactly,
+   * while Any-elements are ignored.
+   */
+  case class Prefix(elems: Prefix.Elem*) {
+
+    def matches(path: Path): Boolean = {
+      if (this.size > path.size)
+        return false
+      var i = 0
+      while (i != this.size)
+        elems(i) match {
+          case Prefix.Label(buf) if buf != path.elems(i) =>
+            return false
+          case _ => // matches
+            i += 1
+        }
+      true
+    }
+
+    // A prefix acts somewhat like a Seq[Elem]
+    def take(n: Int) = Prefix(elems.take(n):_*)
+    def drop(n: Int) = Prefix(elems.drop(n):_*)
+    def ++(that: Prefix): Prefix =
+      if (that.isEmpty) this
+      else Prefix((elems ++ that.elems):_*)
+    def size: Int = elems.size
+    def isEmpty: Boolean = elems.isEmpty
+
+    def ++(path: Path): Prefix =
+      if (path.isEmpty) this
+      else this ++ Prefix(path)
+
+    lazy val showElems = elems.map(_.show)
+    lazy val show = showElems.mkString("/", "/", "")
+    override def toString = s"""Prefix(${showElems.mkString(",")})"""
+  }
+
+  object Prefix {
+
+    def apply(path: Path): Prefix =
+      Prefix(path.elems.map(Label(_)):_*)
+
+    /**
+     * Parse `s` as a prefix matching expression with concrete syntax
+     *
+     * {{{
+     * path       ::= '/' elems | '/'
+     *
+     * elems      ::= elem '/' elem | elem
+     *
+     * elem       ::= '*' | label
+     *
+     * label      ::= (\\x[a-f0-9][a-f0-9]|[0-9A-Za-z:.#$%-_])+
+     *
+     * }}}
+     *
+     * for example
+     *
+     * {{{
+     * /foo/bar/baz
+     * /foo&#47;*&#47;/bar/baz
```

Comment:
I'm having trouble interpreting this, what is this supposed to mean? /foo//*//bar/baz?